File: 80. Remove Duplicates from Sorted Array II.cpp
class Solution {
public:
    // two pointers: 13ms
    // This is a general solution, 2 can be repleaced by any integer k
    // to realize the problem of allowing k duplicates in the sorted array
    int removeDuplicates(vector<int>& nums) {
        if(nums.size() <= 2) return nums.size();
        int index = 2;
        for(int i = 2; i < nums.size(); i++){
            if(nums[i] != nums[index - 2]){
                nums[index] = nums[i];
                index++;
            }
        }
        return index;
    }
};

File: 80. Remove Duplicates from Sorted Array II.py
class Solution(object):
    
    # two pointer: O(N) time, O(1) space
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if nums == []:
            return 0
        i, count = 0, 1
        for j in xrange(1, len(nums)):
            if nums[j] != nums[i]:
                i += 1
                nums[i] = nums[j]
                count = 1
            elif nums[j] == nums[i] and count < 2:
                count += 1
                i += 1
                nums[i] = nums[j]
        return i + 1                    
        